### PR TITLE
Jira integration

### PR DESCRIPTION
**Pull request by mhib**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -20,6 +20,7 @@ gem 'business_time'
 gem 'holidays'
 gem 'sentry-raven'
 gem 'bootstrap-sass'
+gem 'jira-ruby'
```
Comment (kbackowski) [inline Gemfile]: let's add require: false, and load it only where needed (ideally in service class)

File: app/assets/stylesheets/external_auths.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the ExternalAuths controller here.
```
Comment (kbackowski) [inline app/assets/stylesheets/external_auths.scss]: we don't need this file

File: app/assets/javascripts/external_auths.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (kbackowski) [inline app/assets/javascripts/external_auths.coffee]: we don't need this file

File: config/locales/en.yml [hunk 4]
```diff
@@ -110,10 +113,10 @@ en:
           attributes:
             task:
               invalid_uri: 'Task URL is not valid'
+              invalid_external: 'Task URL does not point to a valid external tasak'
```
Comment (kbackowski) [inline config/locales/en.yml]: typo

File: config/locales/pl.yml [hunk 5]
```diff
@@ -72,6 +72,8 @@ pl:
       phone: Telefon
       button_disabled_tooltip: Tworzenie kont zablokowane. Automatyczne tworzenie poprzez logowanie LDAP.
       email_edit_disabled: Edycja adresu email zablokowana. Adres email jest nadawany przez LDAP.
+    external_auths:
+      authorized: 'Zaautoryzowano z'
```
Comment (kbackowski) [inline config/locales/pl.yml]: typo

File: app/javascript/components/time_table/projects/external_auth.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            <a className="btn btn-danger" onClick={this.onDelete}>{I18n.t('common.destroy')}</a>
+          </p>
+        </div>
+      );
+    }
+    return (
+      <div>
+        <h3>New auth</h3>
+        <input placeholder="Domain" value={this.state.domain} onChange={this.onDomainChange}></input>
+        <button onClick={this.getAuthLink}>Generate link</button>
```
Comment (kbackowski) [inline app/javascript/components/time_table/projects/external_auth.js]: Missing translations. Please review all views and add missing translations.

Comment (mhib) [reply app/javascript/components/time_table/projects/external_auth.js]: Fixed

File: app/services/work_time_saver.rb [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    if (payload = validate_integration(work_time, external_auth))
+      save_with_payload(payload, additional_params)
+    else
+      InvalidWorkTime.new(work_time)
+    end
+  end
+
+  private
+
+  class InvalidWorkTime
```
Comment (kbackowski) [inline app/services/work_time_saver.rb]: let's try to solve this in some other way i.e creating from object class with validations which will copy validation errors from WorkTime.

```
class WorkTimeForm
  include ActiveModel::Model
  validate :validates_ticker_url

  def save
    ...
  end

  def valid?
    super
    copy_errors
    errors.empty?
  end

  private

  def copy_errors
    work_time.valid?
    work_time.errors.details.each do |key, value|
      errors.add(key, value.first[:error])
    end
  end
end
```

Comment (mhib) [reply app/services/work_time_saver.rb]: Changed

**Pull request merged**